src/test/java/org/cactoos/experimental/ThreadsTest.java

```diff
@@ -24,114 +24,62 @@
-                        return "1st";
-                    },
-                    () -> {
-                        TimeUnit.SECONDS.sleep(6);
-                        return "3rd";
-                    },
-                    () -> {
-                        TimeUnit.SECONDS.sleep(4);
-                        return "2nd";
-                    }
-                )
-            );
-            MatcherAssert.assertThat(
-                new Mapped<>(Future::get, futures),
-                Matchers.hasItems("1st", "3rd", "2nd")
-            );
-        } catch (final InterruptedException exp) {
-            Thread.currentThread().interrupt();
-            throw new IllegalStateException(exp);
-        } finally {
-            try {
-                extor.shutdown();
-                if (!extor.awaitTermination(2, TimeUnit.SECONDS)) {
-                    extor.shutdownNow();
-                }
-            } catch (final InterruptedException exp) {
-                extor.shutdownNow();
-            }
-        }
-    }
-
     /**
      * Execute the tasks concurrently using {@link Threads} when
      *  {@link ExecutorService} was initiated by someone else.
      */
     @Test
-    public void cactoosWay() {
-        this.repeatWithTimeout(
+    public void containsResults() {
+        this.repeat(
             arg -> {
                 final ExecutorService extor = Executors.newFixedThreadPool(3);
                 try {
-                    MatcherAssert.assertThat(
-                        new Threads<String>(
+                    new Assertion<>(
+                        "contains results from callables",
+                        () -> new Threads<String>(
                             extor,
                             () -> {
-                                TimeUnit.SECONDS.sleep(3);
+                                this.sleep();
                                 return "txt 1";
                             },
                             () -> {
-                                TimeUnit.SECONDS.sleep(3);
+                                this.sleep();
                                 return "txt 2";
                             },
                             () -> {
-                                TimeUnit.SECONDS.sleep(3);
+                                this.sleep();
                                 return "txt 3";
                             }
                         ),
                         new HasValues<>("txt 1", "txt 2", "txt 3")
-                    );
+                    ).affirm();
                 } finally {
-                    try {
-                        extor.shutdown();
-                        if (!extor.awaitTermination(1, TimeUnit.SECONDS)) {
-                            extor.shutdownNow();
-                        }
-                    } catch (final InterruptedException exp) {
+                    extor.shutdown();
+                    if (!extor.awaitTermination(1, TimeUnit.SECONDS)) {
```

Comment:
`1L` if we don't want implicit conversion

Reply:
@ilyakharlamov ok